lib/sqlalchemy/dialects/postgresql/psycopg2.py

```diff
@@ -551,13 +590,55 @@ def _log_notices(self, cursor):
 
 
 class PGCompiler_psycopg2(PGCompiler):
-    pass
+    def __init__(
+        self, dialect, statement, column_keys=None, inline=False, **kwargs
+    ):
+        self.dialect = dialect
+        self.multiple_rows = inline
+        self.execute_values_insert_template = None
+
+        super(
+            PGCompiler_psycopg2,
+            PGCompiler_psycopg2).__init__(
+            self,
+            dialect,
+            statement,
+            column_keys,
+            inline,
+            **kwargs)
+
+    # Override SQLCompiler.generate_values_placeholders_str - enable use of
+    # psycopg2 execute_values()
+    def generate_values_placeholders_str(
+            self, crud_params, returning_clause_exists):
+        # Currently not using psycopg2.execute_values() when there's a returning clause; need to add support
+        # for receiving multiple return values from insert query
+        if self.multiple_rows and not returning_clause_exists and self.dialect.psycopg2_execution_mode == EnumPsycopg2ExecutionMode.VALUES_BATCH:
+            self.execute_values_insert_template = "(" + \
+                ", ".join([c[1] for c in crud_params]) + ")"
+            return " VALUES %s"
+        else:
+            return super(
+                PGCompiler_psycopg2,
+                PGCompiler_psycopg2).generate_values_placeholders_str(
+                self,
+                crud_params,
+                returning_clause_exists)
 
 
 class PGIdentifierPreparer_psycopg2(PGIdentifierPreparer):
     pass
 
 
+class EnumPsycopg2ExecutionMode(Enum):
```

Comment:
we support Python 2, can't use Enum.  Please use util.symbol:

```
from .. import util
SINGLE_STATEMENT = util.symbol("single_statement")
```

Reply:
ok